internal/film/fimlrepo_mongo.go

```diff
@@ -0,0 +1,49 @@
+package film
+
+import (
+	imdb "github.com/PhamDuyKhang/littledetective/internal/pkg/crawler"
+	"github.com/PhamDuyKhang/littledetective/internal/types"
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+)
+
+type (
+	FilmMongoRepository struct {
+		ss *mgo.Session
+	}
+)
+
+func NewFilmMongoRepository(s *mgo.Session) *FilmMongoRepository {
+	return &FilmMongoRepository{
+		ss: s,
+	}
+}
+
+func (frp FilmMongoRepository) InsertFilm(f types.Film) (types.Film, error) {
+	session := frp.ss.Clone()
+	defer session.Clone()
+	f.ID = imdb.NewUUID()
```

Comment:
Should check if `f.ID != ""` first, just in case, the caller of this function already init the ID and keep that data for later use. But inside this func, we changed that data.  
Also, normally, this line of code (LoC) should not stay at the repo layer as it's the business logic that's is not related to inserting document.

Reply:
I'm understood, the logic code should be put in service class